### PR TITLE
Sets default theme to a Hacker Style theme

### LINKED ISSUE #289: Hacker Style Theme
Can we add a default theme similar to Nang's favorite theme, Palm Synthwave?

### PR DESCRIPTION
**Pull request by vedantgrover**
Fixes #289 

This PR adds the palm-wave-ext package to the extensions folder and it adds palm-wave-ext as a built in extension. It also adds further configuration in the workbench to set the theme as the default value for the user. The user can still change the theme if needed.

### CODE REVIEW

**Comment by Fryingpannn** [comment]
yoo, any screenshots? 👀

**Comment by vedantgrover** [comment]
@Fryingpannn I gotchu bro

Screenshot 2024-09-01 at 23 48 48
Screenshot 2024-09-01 at 23 49 38

File: src/vs/workbench/browser/workbench.ts [hunk 1]
```diff
@@ -3,65 +3,104 @@
...
-import { AccessibleViewRegistry } from 'vs/platform/accessibility/browser/accessibleViewRegistry';
-import { NotificationAccessibleView } from 'vs/workbench/browser/parts/notifications/notificationAccessibleView';
+import "vs/workbench/browser/style";
+import { localize } from "vs/nls";
+import { runWhenWindowIdle } from "vs/base/browser/dom";
+import {
+	Event,
+	Emitter,
+	setGlobalLeakWarningThreshold,
+} from "vs/base/common/event";
```
Comment (Himanshu-Singh-Chauhan) [inline src/vs/workbench/browser/workbench.ts:13]: don't change formatting, check your eslint or save without formatting from command pallete

File: product.json [hunk 2]
```diff
@@ -2321,6 +2323,12 @@
 	},
 	"chatParticipantRegistry": "https://az764295.vo.msecnd.net/extensions/chat.json",
 	"builtInExtensions": [
+		{
+			"name": "palm-code.palm-synthwave",
+			"version": "0.0.3",
```
Comment (Himanshu-Singh-Chauhan) [inline product.json:2328]: does this download it automatically from marketplace?

File: src/vs/workbench/contrib/preferences/common/preferencesContribution.ts [hunk 3]
```diff
@@ -143,24 +224,41 @@ export class PreferencesContribution implements IWorkbenchContribution {
...
+			default: "filter",
+			scope: ConfigurationScope.WINDOW,
 		},
-	}
+		// "workbench.colorTheme": {
+		// 	type: "string",
+		// 	default: "palm-wave-ext",
+		// 	description: "Specifies the color theme used in the workbench.",
+		// },
+	},
```
Comment (Himanshu-Singh-Chauhan) [inline src/vs/workbench/contrib/preferences/common/preferencesContribution.ts:263]: probably this can be removed

File: product.json [hunk 4]
```diff
@@ -2321,6 +2323,12 @@
...
+			"publisherDisplayName": "Palm Code",
+			"repo": "https://github.com/RobertoGarc/palm-wave-ext/tree/main"
```
Comment (Himanshu-Singh-Chauhan) [inline product.json:2330]: the repo is open source, but can we ship this theme in our app??
also we default theme might the pear theme that tanishq is working on.
gotta confirm with nang

**Review by Himanshu-Singh-Chauhan — CHANGES_REQUESTED** [review]

**Comment by vedantgrover** [comment]
Just FYI, I didn't have to configure anything. This is the default theme.

Not sure if the Issue wanted the **option** to switch to this theme or to use the theme as default. I think it's better to have it as default as it adds a bit of character and uniqueness to the app.

File: product.json (hunk 2 above)
Comment (vedantgrover) [reply product.json:2328]: Yep

**Comment by nang-dev** [comment]
this is good, but going to close this. I probably wasn't clear enough, but adding the extension is a bit too much. The color scheme is just a bunch of hex values, so we just want to hard code it in like:

https://github.com/trypear/pearai-submodule/pull/191

thanks sm though vedant, really appreciate this

**Pull request closed without merging**